testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -1015,6 +1015,24 @@ public void testStringPartitioningWithSpecialCharactersInsertInSpark()
         onTrino().executeQuery("DROP TABLE " + trinoTableName);
     }
 
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testPartitioningWithMixedCaseColumnUnsupportedInTrino()
+    {
+        String baseTableName = "test_partitioning_with_mixed_case_column_in_spark";
+        String trinoTableName = trinoTableName(baseTableName);
+        String sparkTableName = sparkTableName(baseTableName);
+
+        onSpark().executeQuery("DROP TABLE IF EXISTS " + sparkTableName);
+        onSpark().executeQuery(format(
+                "CREATE TABLE %s (id INTEGER, `mIxEd_COL` STRING) USING ICEBERG",
+                sparkTableName));
+        assertQueryFailure(() -> onTrino().executeQuery("ALTER TABLE " + trinoTableName + " SET PROPERTIES partitioning = ARRAY['mIxEd_COL']"))
```

Comment:
This is a bummer.
We should provide in Trino a way for the users to cope with such situations because otherwise the users would face a Spark lock-in for such situations.

I've created a PR in iceberg to address this problem.  https://github.com/apache/iceberg/pull/5110

Reply:
To give you some context. The original PR I did handle this correctly. According SQL quoted identifier semantics, a quoted identifier should be matched case sensitively. The backticks in Spark behave the same as the quoted identifiers in SQL. IMHO this is not an issue with Iceberg.

There was some discussion about this feature as indeed in Trino the column names are converted into lowercase. Take for example this query

```
CREATE TABLE t("X" bigint) WITH (partitioning = ARRAY['"X"']); 
```

Because the column name is converted to lowercase in Trino, this query would fail, as at that time the "X" has become x and the partitioning parsing logic fails to find this column. This is definitely confusing for the user. In a ALTER TABLE scenario however this is not true. The column will be known as "X". So we agreed on blocking this scenario for now, as mentioned on https://github.com/trinodb/trino/issues/12226#issuecomment-1128839259

Here is the code that explicitly blocks this, removing this would fix the Trino query above.

https://github.com/trinodb/trino/blob/2dd644fc00c636ecae168c81644761c44101327d/plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergUtil.java#L336-L343

Reply:
> IMHO this is not an issue with Iceberg.

I think that the PR https://github.com/apache/iceberg/pull/5110 is more of a usability "improvement" and not a bug. 

Please correct me if I'm wrong, I don't think it should be mandatory to specify the source column name in the same case in the table definition.

Reply:
I think we better stick to SQL semantics.

Imagine following query perfectly valid syntax (not currently working in Trino, but actually working on snowflake):

<img width="630" alt="image" src="https://user-images.githubusercontent.com/9265503/174984538-b5fd4de9-1703-4d48-88b8-fb4c3a4a5fca.png">


```
CREATE TABLE t("X" bigint, "x" bigint) WITH (partitioning = ARRAY['"X"']); 
```

Because of the quoted identifers we would know which x to match. I would think Iceberg partitioning spec should be exactly matched against the Iceberg schema and not impose a certain way of working.